transport/whisper/types/filter.go

```diff
@@ -0,0 +1,24 @@
+package whispertypes
+
+import (
+	"crypto/ecdsa"
+)
+
+// Filter represents a Whisper message filter
+type Filter interface {
+	Src() *ecdsa.PublicKey      // Sender of the message
+	KeyAsym() *ecdsa.PrivateKey // Private Key of recipient
+	KeySym() []byte             // Key associated with the Topic
+	Topics() [][]byte           // Topics to filter messages with
+	PoW() float64               // Proof of work as described in the Whisper spec
+	AllowP2P() bool             // Indicates whether this filter is interested in direct peer-to-peer messages
+	SymKeyHash() Hash           // The Keccak256Hash of the symmetric key, needed for optimization
```

Comment:
Why is `SymKeyHash` here? Should this not be a whisper implementation specific internal?

Reply:
To be honest, I did a mostly automated search and replace over the structs, so everything is present here. Now that everything is building and working again, I can go over these methods and remove the ones which are never used.